src/Diagnostics.Scripts/EntityInvoker.cs

```diff
@@ -395,25 +396,32 @@ private void Validate()
                     throw new ScriptCompilationException("Detector is marked with both SystemFilter and ResourceFilter. System Invoker should not include any ResourceFilter attribute.");
                 }
 
-                if((this._entityMetaData.ScriptText.Contains("All('") || this._entityMetaData.ScriptText.Contains("All(\"")) && 
+                if (this._entityMetaData.ScriptText.Contains("SuppressMessage(", StringComparison.OrdinalIgnoreCase))
+                {
+                    Regex suppressMessageRegex = new Regex(@"\[(.*)SuppressMessage\((.*)\)\]", RegexOptions.IgnoreCase);
+                    if (suppressMessageRegex.Match(this._entityMetaData.ScriptText).Success)
+                    {
+                        throw new ScriptCompilationException("ERROR : Use of System.Diagnostics.CodeAnalysis.SuppressMessage attribute is prohibited.");
+                    }
+                }
+
+                if (this._entityMetaData.ScriptText.Contains("pragma", StringComparison.OrdinalIgnoreCase))
```

Comment:
@nmallick1 remind me again what one can do with pragma?

Reply:
@hforeste You can configure pragma tag to suppress code analysis generated errors and warnings. A malicious user can use pragma to suppress the error we generate to block restricted API's. Once the error is suppressed, they are free to use any method calls. Essentially use pragma to mimic the behavior of he SuppressMessage attribute.